keps/sig-release/2853-kubernetes-kubernetes-repository-branch-rename/README.md

```diff
@@ -0,0 +1,777 @@
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+- We aim to make the change during the start of v1.25 release (spring 2022).
+- Perform a Survey to gather information on downstream consumers and how this might affects their workflow. See [Survey Questions](#survey-questions).
+- Announce the change in the kubernetes-dev and kubernetes-announcement mailing list and in a blog post. As well as use Twitter and other social media to announce.
+- Change all open Pull Requests to `Draft mode` (so that when we rename the branch, tests will not be triggered, and we avoid a massive queue in the CI infrastructure).
+- Update all Prow jobs that use `kubernetes/kubernetes` and use the `master` branch to listen to the `main` branch as well
+- Follow the guide on https://github.com/kubernetes/community/blob/master/github-management/default-branch-migration.md to rename the repository branch name.
+- After the branch rename, announce that in the mailing list
+- Update all Prow jobs that use `kubernetes/kubernetes` and to remove the `master` branch
+- Convert on demand the open Pull Requests from the `Draft mode` to `Review ready mode`
+
+
+References:
+- https://github.com/kubernetes/org/issues/2222
+- https://github.com/kubernetes/enhancements/issues/2853
+- https://github.com/kubernetes/kubernetes/issues/105601
+- https://github.com/kubernetes/community/blob/master/github-management/default-branch-migration.md
+
+- https://github.com/github/renaming
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+As a contributor, I have to remember to use `master` when working with repositories that
+haven't yet switched.
+Switching k/k to `main` will reduce the burden slightly.
+
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
```

Comment:
Add a risk that we may break downstream consumers, we use the survey as mitigation as well as the schedule to let them prepare their build systems for the change.